src/main/scala/cromwell/engine/backend/jes/JesAttachedDisk.scala

```diff
@@ -0,0 +1,89 @@
+package cromwell.engine.backend.jes
+
+import com.google.api.services.genomics.model.Disk
+import cromwell.engine.Hashing._
+import cromwell.engine.backend.runtimeattributes.DiskType
+import wdl4s.ThrowableWithErrors
+
+import scala.util.Try
+import scalaz.Scalaz._
+import scalaz._
+
+object JesAttachedDisk {
+  val Identifier = "[a-zA-Z0-9-_]+"
+  val Directory = """/[^\s]+"""
+  val Integer = "[0-9]+"
+  val WorkingDiskPattern = s"""${JesWorkingDisk.Name}\\s+($Integer)\\s+($Identifier)""".r
+  val MountedDiskPattern = s"""($Directory)\\s+($Integer)\\s+($Identifier)""".r
+
+  def parse(s: String): Try[JesAttachedDisk] = {
+    val validation = s match {
+      case WorkingDiskPattern(sizeGb, diskType) =>
+        (validateLong(sizeGb) |@| validateDiskType(diskType)) { (s, dt) =>
+          JesWorkingDisk(dt, s.toInt)
+        }
+      case MountedDiskPattern(mountPoint, sizeGb, diskType) =>
+        (validateLong(sizeGb) |@| validateDiskType(diskType)) { (s, dt) =>
+          JesEmptyMountedDisk(dt, s.toInt, mountPoint)
+        }
+      case _ => s"Disk strings should be of the format 'local-disk SIZE TYPE' or '/mount/point SIZE TYPE'".failureNel
+    }
+
+    Try(validation match {
+      case Success(localDisk) => localDisk
+      case Failure(nels) =>
+        throw new UnsupportedOperationException with ThrowableWithErrors {
+          val message = ""
+          val errors = nels
+        }
+    })
+  }
+
+  private def validateDiskType(diskTypeName: String): ValidationNel[String, DiskType] = {
+    DiskType.values().find(_.diskTypeName == diskTypeName) match {
+      case Some(diskType) => diskType.successNel[String]
+      case None =>
+        val diskTypeNames = DiskType.values.map(_.diskTypeName).mkString(", ")
+        s"Disk TYPE $diskTypeName should be one of $diskTypeNames".failureNel
+    }
+  }
+
+  private def validateLong(value: String): ValidationNel[String, Long] = {
+    try {
+      value.toLong.successNel
+    } catch {
+      case _: IllegalArgumentException => s"$value not convertible to a Long".failureNel[Long]
+    }
+  }
+}
+
+trait JesAttachedDisk {
+  def name: String
+  def diskType: DiskType
+  def sizeGb: Int
+  def mountPoint: String
+  def toGoogleDisk: Disk = {
+    new Disk().setName(name)
+      .setType(diskType.googleTypeName)
+      .setAutoDelete(true)
+      .setSizeGb(sizeGb)
+      .setMountPoint(mountPoint)
+  }
+}
+
+case class JesEmptyMountedDisk(diskType: DiskType, sizeGb: Int, mountPoint: String) extends JesAttachedDisk {
+  val name = mountPoint.md5Sum.substring(0, 8)
+  override def toString: String = s"$mountPoint $sizeGb ${diskType.diskTypeName}"
+}
+
+object JesWorkingDisk {
+  val MountPoint = "/cromwell_root"
```

Comment:
Nit pick: Could that be a path initially ? It looks like there's a lot of Paths.get(JesWorkingDisk.MountPoint) that might be avoided if this was already a path. If it's too annoying to change don't bother though.